uuid.go

```
@@ -56,7 +56,9 @@ func UUIDNonEmpty(ctx *Ctx, args ...string) (h string) {
 		cmdLine := []string{"uuid.py", "a"}
 		cmdLine = append(cmdLine, args...)
 		h, _, err = ExecCommand(ctx, cmdLine, "", nil)
-		FatalOnError(err)
+		if err != nil {
+			return ""
+		}
 		h = h[:len(h)-1]
```

So we ignore the error (I don't think there should be an error in UUID generation) and return an empty UUID instead? That will be wrong, UUID must be non-empty and fixed length.

---

This one should no longer used, all code should reference uuid module of libraries repo

---

No, this `if branch` *is* and *will be* used when legacy mode *is* selected. Your branch (UUID from libraries is used by default in both affs UUID and document UUID).